File: python/977A_WrongSubtraction.py
#!/usr/bin/env python3

i, n = list(map(int,input().split()))

while n > 0:
    c = i % 10
    if c == 0:
        i //= 10
        n -= 1
    else:
        c = min(c, n)
        i -= c
        n -= c

print(i)

File: cpp/977A_WrongSubtraction.cpp
#include <iostream>
using namespace std;
int main() {
    int i, n;
    cin >> i;
    cin >> n;

    while (n > 0) {
        int c = i % 10;
        if (c == 0) {
            i /= 10;
            n--;
        } else {
            c = min(c, n);
            i -= c;
            n -= c;
        }
    }

    cout << i;
}